toolkit/tools/internal/file/file_test.go

```diff
@@ -123,3 +125,115 @@ func TestRemoveDirectoryContentsNonExistent(t *testing.T) {
 	err = RemoveDirectoryContents(tempDir)
 	assert.Error(t, err)
 }
+
+func TestCopyDir(t *testing.T) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		logger.Log.Panicf("Failed to get working directory, error: %s", err)
+	}
+	testDir := filepath.Join(workingDir, "testdata")
+
+	// Defining and creating src directory
+	src := testDir + "/source"
+	err = os.MkdirAll(src, os.ModePerm)
+	assert.NoError(t, err)
+
+	// Adding test files into src directory
+	err = CreateTestFiles("testfile", src)
+	assert.NoError(t, err)
+
+	// Defining dst directory and child permissions
+	newDirPermissions := fs.FileMode(0755)
+	mergeDirPermissions := fs.FileMode(0755)
+	childFilePermissions := fs.FileMode(0755)
+
+	// Defining dst directory and copying src into dst
+	dst := testDir + "/destination"
+	err = CopyDir(src, dst, &newDirPermissions, &mergeDirPermissions, &childFilePermissions)
+	assert.NoError(t, err)
+
+	// verifying the directories are equal
+	equal, err := AreDirectoriesEqual(src, dst)
+	assert.NoError(t, err)
+	assert.True(t, equal)
+
+	// Removing all test files and directories
+	err = os.RemoveAll(testDir)
+	assert.NoError(t, err)
+}
+
+func CreateTestFiles(filename string, outputDir string) error {
```

Comment:
nit: Rename to `createTestFiles` to make the function private. Same thing for the other helper functions.

Reply:
Fixed this.